views/default/elgg/Ajax.js

```diff
@@ -124,17 +145,26 @@ define(function (require) {
 		 *
 		 * See fetch() for more options.
 		 *
-		 * @param {String} path    URL path. e.g. "foo/bar"
+		 * @param {String} path    path or URL. e.g. "foo/bar"
 		 * @param {Object} options {@link jQuery#ajax}. See fetch()
 		 * @returns {jqXHR}
 		 */
 		this.path = function (path, options) {
 			elgg.assertTypeOf('string', path);
 
+			// https://example.org/elgg/foo/?arg=1#bar => foo/?arg=1
+			if (path.indexOf(site_url) === 0) {
```

Comment:
I think we should bail if it's an external URL. I don't think we should encourage use of this API for fetching external resources.